Models/articles.models.js

```
@@ -0,0 +1,13 @@
+const db = require("../db/connection");
+
+exports.queryByArticleId = function (param) {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1", [param.article_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 400, msg: "Article does not exist" });
+      } else {
+        return rows[0];
```

Instead of checking the length and returning a rejected promise, PSQL will return an error if the articleID is not a valid data type which is whar you were trying to test for this I believe.